File: nostos-core/src/test/java/sncf/oui/nostos/core/coordinates/CoordinatesTest.java
package sncf.oui.nostos.core.coordinates;

import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.CsvSource;

import static org.assertj.core.api.Assertions.assertThat;

public class CoordinatesTest {


    @ParameterizedTest
    @CsvSource(delimiter = ';', value = {"0.;0.", "-89.;0.", "89.;0.", "0.;-179.", "0.;179."})
    void should_return_latitude_and_longitude(final double latitude, final double longitude) {
        final Coordinates coordinates = Coordinates.of(latitude, longitude);
        assertThat(coordinates.getLatitude()).isEqualTo(latitude);
        assertThat(coordinates.getLongitude()).isEqualTo(longitude);
    }

}

File: nostos-core/src/test/kotlin/io/fries/nostos/core/coordinates/CoordinatesTest.kt
package io.fries.nostos.core.coordinates

import net.jqwik.api.ForAll
import net.jqwik.api.Property
import org.assertj.core.api.Assertions.assertThat

internal class CoordinatesTest {

    @Property
    internal fun should_create_coordinates_from_degrees(
            @ForAll @LatitudeDegrees latitude: Double,
            @ForAll @LongitudeDegrees longitude: Double
    ) {
        val coordinates = Coordinates.of(latitude, longitude)

        assertThat(coordinates.latitude).isEqualTo(latitude)
        assertThat(coordinates.longitude).isEqualTo(longitude)
    }
}